contentcuration/contentcuration/frontend/shared/vuex/session/index.js

```diff
@@ -54,23 +57,34 @@ export default {
   currentLanguage: Languages.get(langCode(window.languageCode || 'en')),
   currentChannelId: window.channel_id || null,
   mutations: {
-    UPDATE_CURRENT_USER(state, userData) {
+    ADD_SESSION(state, currentUser) {
+      state.currentUser = currentUser;
+    },
+    UPDATE_SESSION(state, data) {
       state.currentUser = {
         ...state.currentUser,
-        ...userData,
+        ...data,
       };
     },
-    SET_CURRENT_USER(state, currentUser) {
-      state.currentUser = {
-        ...currentUser,
-      };
-      state.loggedIn = Boolean(currentUser);
+    REMOVE_SESSION(state) {
+      state.currentUser = GUEST_USER;
+      // prevent from double reloading of the login page
+      // ('REMOVE_SESSION' might be triggered by dexie-observable
+      // when reseting IndexedDB)
+      if (!window.location.pathname.endsWith(window.Urls.accounts())) {
```

Comment:
This feels outside of the scope of a Vuex mutation - presumably the goal here is to redirect whenever there is a change to the current user, but I don't think we should do it inside of mutation.

I would suggest either adding a vuex watcher to handle this, or it may be simpler to add a listener for the delete event directly to the Session Resource. There's an example of this for the Task Resource: https://github.com/learningequality/studio/blob/develop/contentcuration/contentcuration/frontend/shared/data/resources.js#L1410

Reply:
Oh, that's nice. I was not too fond of that place either.